commonskmm/src/commonMain/kotlin/org/dhis2/mobile/commons/model/CustomIntentModel.kt

```diff
@@ -10,7 +10,7 @@ data class CustomIntentModel(
 
 data class CustomIntentRequestArgumentModel(
     val key: String,
-    val value: String,
+    val value: Any,
```

Comment:
Wouldn't it be better to use a wildcard instead of Any?

Reply:
What do you mean? I think that having `val value: *` is not valid. How would it be?

Reply:
If you do so, then I guess the model will have to be:
```
data class CustomIntentModel(
  ...
  val customIntentRequest: List<CustomIntentRequestArgumentModel<*>>,
)
```
Each request argument might have a different type.

That is valid as well, but I am not sure about the advantage. If you usually do it this way in the app, that's fine, up to you.

Reply:
I have been looking and it seems that `value: Any` is not type safe so you need to perform manual casting what can lead to runtime errors. `value: T` is type safe in compilation time but adds complexity when serializing or deserializing.
As we don't need to deserialize the object, because we map it, I think that it's better to use `value: T`